Packages/com.unity.inputsystem/InputSystem/Actions/InputActionSetupExtensions.cs

```diff
@@ -1232,6 +1234,49 @@ public BindingSyntax WithInteractions(string interactions)
                 return this;
             }
 
+            /// <summary>
+            /// Remove the last instance of the matching interaction type from the list of interactions.
+            /// </summary>
+            /// <typeparam name="TInteraction"></typeparam>
+            /// <returns>The same binding syntax for further configuration.</returns>
+            /// <exception cref="InvalidOperationException"></exception>
+            /// <remarks>
+            /// Be aware that this method will allocate managed memory due to interaction string parsing.
+            /// </remarks>
+            public BindingSyntax RemoveInteraction<TInteraction>() where TInteraction : IInputInteraction
+            {
+                if (!valid)
+                    throw new InvalidOperationException("Accessor is not valid");
+
+                var interactionName = InputInteraction.s_Interactions.FindNameForType(typeof(TInteraction));
+                if (interactionName.IsEmpty())
+                    throw new NotSupportedException($"Type '{typeof(TInteraction)}' has not been registered as a interaction");
+
+                var interactionsString = m_ActionMap.m_Bindings[m_BindingIndexInMap].interactions;
+                if (string.IsNullOrEmpty(interactionsString))
+                    return this;
+
+                var nameAndParameters = NameAndParameters.ParseMultiple(interactionsString);
+                var interactions = nameAndParameters as List<NameAndParameters> ?? nameAndParameters.ToList();
+                var lastIndexOfInteraction = -1;
+                for (var i = interactions.Count - 1; i >= 0; i--)
+                {
+                    if (interactions[i].name == interactionName)
+                    {
+                        lastIndexOfInteraction = i;
```

Comment:
Do we allow the interaction to be present multiple times?
If its present multiple times should we remove all?

(+ Should we prevent multiple interactions of the same name in the add code?
 I guess we don't know what impact that would have for some odd user custom interactions. )

Reply:
XR Interaction Toolkit ships an input actions asset with a binding with multiple interactions of the same type (but with different properties set).

Reply:
I think `RemoveInteraction` is not actually needed by this PR, so I suggest we remove it entirely here and only add it if we have a clear need to have it and design that resolves these issues.